File: JPAQuickFix/src/main/java/com/skilldistillery/quickfix/entities/BidComment.java
package com.skilldistillery.quickfix.entities;

import java.time.LocalDateTime;
import java.util.Objects;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name= "bid_comment")
public class BidComment {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	
	private String content;
	
	@Column(name= "comment_date")
	private LocalDateTime commentDate;
	
	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getContent() {
		return content;
	}

	public void setContent(String content) {
		this.content = content;
	}

	public LocalDateTime getCommentDate() {
		return commentDate;
	}

	public void setCommentDate(LocalDateTime commentDate) {
		this.commentDate = commentDate;
	}


	public BidComment() {
		super();
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		BidComment other = (BidComment) obj;
		return id == other.id;
	}

	@Override
	public String toString() {
		return "BidComment [id=" + id + ", content=" + content + ", commentDate=" + commentDate + "]";
	}
	
	
	
}

File: ngQuickFix/src/app/models/bid-comment.ts
import { Bid } from "./bid";
import { User } from "./user";

export class BidComment {
    id: number;
    content: string;
    commentDate: string;
    bid: Bid;
    user: User;
    
    constructor(
    id: number = 0,
    content: string = '',
    commentDate: string = '',
    bid: Bid = new Bid(),
    user: User = new User(),
    ){
    this.id = id;
    this.content = content ;
    this.commentDate = commentDate;
    this.bid = bid;
    this.user = user;
    }


}
